dali/pipeline/operator/builtin/caching_list.h

```diff
@@ -108,22 +108,26 @@ class CachingList {
   const T &PeekProphet() {
     if (prophet_ == full_data_.end())
       throw std::out_of_range(
-              "Attempted to peek element that doesn't exist. Add more elements to CachingList "
-              "before calling PeekProphet. Even the prophet can't see outside the event horizon.");
+              "Attempted to peek batch data that doesn't exist. Add more elements the DALI"
+              " input operator.");
```

Comment:
```suggestion
              "Attempted to peek the data batch that doesn't exist. Add more elements to the DALI"
              " input operator.");
```
My only concern here is whether this error message would be always helpful. I mean, is it possible that after receiving this error message, adding more elements to the DALI input operator won't solve the error? If so, maybe this message should be rephrased to something like `Consider adding more elements [...]`?

Reply:
I'm afraid that now it talks about CachingList that is an implementation details, and prophet what make the user super confused. I'm open to suggestion how we can improve it further.